### DISCUSSION TITLE
imx477 (RaspPi HQ Camera) - Green picture ???

### DISCUSSION THREAD

**Discussion by Coyote-89** [aaronwmorris/indi-allsky#1730] [Q&A]
Hello everyone,

I've just finished my installation and I'm having a problem with the captured images.
They are all green!

If I view the stream in real time (via libcamera-still -t0 ) 
The image is correct.
![Screenshot 2024-12-23 150226](https://github.com/user-attachments/assets/4eebfb1c-fbb0-4d02-a3f5-85ff9028b782)

But since indi: it's all green and I can't identify the parameter that produces this?
![Screenshot 2024-12-23 151150](https://github.com/user-attachments/assets/ec32008e-252b-44d3-9c8c-69abc7de0d98)

If you have any idea.

**Comment by aaronwmorris** [comment]
This is expected behavior due to indi-allsky processing RAW (DNG) data by default.  This just means no white balancing is applied.  There are two green pixels for every red and blue pixel on the sensor which leads to a green bias in the image.

You can return libcamera-still to the default functionality by enabling the AWB settings in the libcamera settings area.  Unfortunately, there is a side effect to enabling AWB.  libcamera implements AWB by taking multiple exposures to figure out the proper balancing settings (for every frame).  At night, this can mean a 30s nighttime exposure can take 2-3 minutes to complete.  You do not notice during the day because exposures are so short.

The alternative is to use one of the SCNR algorithms on the Image tab.  These algorithms are a quick way to remove the green bias from the RAW images.

**Reply by Coyote-89** [reply]
Thanks for your reply. 
Its better with SCNR algorithms ;-)
